docs/override_policy.md

```diff
@@ -0,0 +1,120 @@
+# Override Policy
+
+The [OverridePolicy][1] and [ClusterOverridePolicy][2] are used to declare override rules for resources when 
+they are propagating to different clusters.
+
+## Difference between OverridePolicy and ClusterOverridePolicy
+TBD
+## Resource Selector
+TBD
+## Target Cluster
+TBD
+## Overriders
+
+Karmada offers various alternatives to declare the override rules:
+- `ImageOverrider`: dedicated to varying images for workloads.
+- `CommandOverrider`: dedicated to varying commands for workloads.
+- `ArgsOverrider`: dedicated to varying args for workloads.
+- `PlaintextOverrider`: a general-purpose tool to varying any kind of resources.
+
+### ImageOverrider
+TBD
+### CommandOverrider
+The `CommandOverrider` is a refined tool to override commands(e.g.`/spec/template/spec/containers/0/command`) 
+for workloads, such as `Deployment`.
+
+The allowed operations are as follows:
+- `add`: append one or more flags to the command list.
+- `remove`: removes one or more flags from the command list.
+
+#### Examples
+Suppose we have a deployment named `myapp` as follows.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: myapp
+  ...
+spec:
+  template:
+    spec:
+      containers:
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter1=foo
+            - --parameter2=bar
+```
+
+**Example 1: Add flags when workloads propagating to specific clusters.**
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: OverridePolicy
+metadata:
+  name: example
+spec:
+  ...
+  overriders:
+    commandOverrider:
+      - containerName: alpine
+        operator: add
+        value:
+          - --cluster=member1
+```
+It means `add`(append) a new flag `--cluster=member1` to the `myapp`.
+
+After the policy is applied for `myapp`, the command list will be:
+```yaml
+      containers:
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter1=foo
+            - --parameter2=bar
+            - --cluster=member1
+```
+
+**Example 2: Remove flags when workloads propagating to specific clusters.**
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: OverridePolicy
+metadata:
+  name: example
+spec:
+  ...
+  overriders:
+    commandOverrider:
+      - containerName: alpine
+        operator: remove
+        value:
+          - --parameter1=foo
+```
+It means `remove` the flag `--parameter1=foo` from the command list.
+
+After the policy is applied for `myapp`, the `command` will be:
+```yaml
+      containers:
+        - image: myapp
+          name: myapp
+          command:
+            - ./myapp
+            - --parameter2=bar
+```
+
+### ArgsOverrider
+The `ArgsOverrider` is a refined tool to override args(e.g.`/spec/template/spec/containers/0/args`) for workloads,
```

Comment:
The `ArgsOverrider` is a refined tool to override args (such as `/spec/template/spec/containers/0/args`) for workloads,